### PR TITLE
Update DEVELOPING.md

### PR DESCRIPTION
**Pull request by dyfer**
Add information about accessing AppVeyor builds

## Purpose and Motivation

AppVeyor builds are stpored independently from Travis builds uploaded to Amazon S3. This PR adds instructions how to access these builds, either from a PR page, or from AppVeyor's history.

## Types of changes

- Documentation
## To-do list

- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -114,6 +114,14 @@ are at:
 A build for a specific commit may not always be available: for instance, if the build was cancelled early or failed to
 complete.
 
+AppVeyor builds
+---------------
+Independently from Amazon S3 hosting service, AppVeyor stores its Windows builds temporarily for 6 months. In contrast to the Amazon S3 hosting, these builds are available for _all_ pull requests. The URL to these builds includes AppVeyor's job number and is unrelated to commit SHA, e.g.
+
+* https://ci.appveyor.com/api/buildjobs/76sh4cu20rh50bp1/artifacts/art_folder.zip
```
Comment (snappizz) [inline DEVELOPING.md]: thanks! tiny formatting nitpick -- this should be a code block rather than a single bullet point

Comment (dyfer) [reply DEVELOPING.md]: Thanks @snappizz 
If I change it to be a code block, it stops being a URL. Is that OK?

Comment (snappizz) [reply DEVELOPING.md]: yeah, it's just for display purposes to demonstrate what the URL looks like, right?

**Comment by dyfer** [comment]
@snappizz thank you for your feedback. Please note that I now added information about AppVeyor's history page (which I didn't know about when making this PR).

**Pull request merged**